solidity/contracts/token/libs/MovableCollateralRouter.sol

```diff
@@ -113,20 +117,45 @@ abstract contract MovableCollateralRouter is FungibleTokenRouter {
         uint256 amount,
         ITokenBridge bridge
     ) external payable onlyRebalancer onlyAllowedBridge(domain, bridge) {
-        address rebalancer = _msgSender();
+        bytes32 recipient = _recipient(domain);
+
+        Quote[] memory quotes = bridge.quoteTransferRemote(
+            domain,
+            recipient,
+            amount
+        );
+
+        uint256 collateralFee = 0;
+        for (uint256 i = 0; i < quotes.length; i++) {
+            if (quotes[i].token == token()) {
+                collateralFee += quotes[i].amount;
+            }
+        }
+
+        // charge the rebalancer any bridging fees denominated in the collateral
+        // token to avoid undercollateralization
+        if (collateralFee > amount) {
+            _transferFromSender(collateralFee - amount);
```

Comment:
Maybe this is done elsewhere, but shouldn't we just take `collateralFee` from the rebalancer in all cases?

Reply:
the collateral fee (and quote) will include the amount being bridged, which we want to use this contract's balance for

Reply:
I think this is the superior UX
otherwise the client needs to implement its own logic for deciding when to approve and how much